src/app/loadout-builder/mod-utils.ts

```diff
@@ -141,15 +134,26 @@ export function assignModsToArmorSet(
  * Checks to see if some mod in a collection of LockedMod or LockedMod,
  * has an elemental (non-Any) energy requirement
  */
-export function someModHasEnergyRequirement(mods: LockedMod[]) {
+export function someModHasEnergyRequirement(mods: PluggableInventoryItemDefinition[]) {
   return mods.some(
-    (mod) =>
-      !mod.modDef.plug.energyCost || mod.modDef.plug.energyCost.energyType !== DestinyEnergyType.Any
+    (mod) => !mod.plug.energyCost || mod.plug.energyCost.energyType !== DestinyEnergyType.Any
   );
 }
 
-/** Sorts PluggableInventoryItemDefinition's by energyType else energyCost else name. */
+/**
+ * Sorts PluggableInventoryItemDefinition's by the following list of comparators.
+ * 1. The known plug category hashes, see ./types#knownModPlugCategoryHashes for ordering
+ * 2. itemTypeDisplayName, so that legacy and combat mods are ordered alphabetically by their category name
+ * 3. energyType, so mods in each category go Any, Arc, Solar, Void
+ * 4. by energy cost, so cheaper mods come before more expensive mods
+ * 5. by mod name, so mods in the same category with the same energy type and cost are alphabetical
+ */
 export const sortMods = chainComparator<PluggableInventoryItemDefinition>(
+  compareBy((mod) => {
```

Comment:
Added some extra sorting conditions here so that mod order matches the order of the mod picker sections. Note that it kinda matches the logic in sortModGroups.